application.go

```diff
@@ -431,12 +429,14 @@ func (r *marathonClient) WaitOnApplication(name string, timeout time.Duration) e
 			flick.SwitchOn()
 		}()
 		for !flick.IsSwitched() {
-			if found, err := r.HasApplication(name); err != nil {
-				continue
-			} else if found {
-				if app, err := r.Application(name); err == nil && app.AllTaskRunning() {
-					return nil
+			if err := r.HasApplication(name); err != nil {
+				if err != ErrDoesNotExist {
+					return err
```

Comment:
Unfortunately it's not tested. This method requires more sophisticated behaviour for test than we can provide with endpoint mocks. I'll try to look into it soon.